axonserver/src/main/java/io/axoniq/axonserver/rest/BackupInfoRestController.java

```diff
@@ -39,7 +40,8 @@
 @RestController
 @RequestMapping("/v1/backup")
 public class BackupInfoRestController {
```

Comment:
Thought: would it make sense to introduce some form of spring filters to add logging on that level? It is a form of cross-cutting concern we try to introduce, so off loading it to something like a filter/interceptor might make things more readable.

Reply:
Wouldn't that also require us to annotate all these locations with information on how important the event is? Or what type of event? Don't forget this is not regular logging to trace what happened; this is logging to ensure an audit trail of (security/privacy related) important events exists. Also, you have to ensure no data is inadvertently revealed by the logging that may not be used by persons looking at these logs, while being complete enough to allow for a post-mortem on a security/data breach incident.

The academic in me is tickled pink by the meta-programming and data classification involved, but I'm not sure if that would be less work.

Reply:
I also consider cleaner to use an interceptor, you don't need to annotate all rest endpoints but only to config it. The interceptor can log the request name/path and optionally all the parameters. If you want to avoid logging some of them you can annotate only these fields, for example. The centralization of this kind of logging allows you, for example, to change more easily the prefix principal@context if you need it in the future.  Also, in this way we will not forget to add audit log to the API implemented in the future :)